Added Function field in Edit Panel

SeemaKumaran opened the pull request:
Now use Metric field in the Edit Panel only for Metrics. To add function
along with the metrics, use Function field in the Edit panel. This is
fixed.

SeemaKumaran committed:
Added Function field in Edit Panel

Now use Metric field in the Edit Panel only for Metrics. To add function
along with the metrics, use Function field in the Edit panel. This is
fixed.

alexanderzobnin commented on src/query_ctrl.js:
```diff
@@ -144,6 +160,15 @@ export class MQEQueryCtrl extends QueryCtrl {
   }
 
+  getFunctions() {
+    var _this10 = this;
```
Why `_this10`? Strange name.

alexanderzobnin commented on src/query_ctrl.js:
```diff
@@ -144,6 +160,15 @@ export class MQEQueryCtrl extends QueryCtrl {
+  getFunctions() {
+    var _this10 = this;
+
+    return this.exploreMetrics('functions').then(function (functions) {
```
This plugin use ES6, so you can use arrow function to make it simpler:

```
return this.exploreMetrics('functions').then((functions) => {
```

By using arrow functions you don't need to reassign `this`, because arrow function hasn't its own `this`.
So you can write this:

```
getFunctions() {
  return this.exploreMetrics('functions').then((functions) => {
    var segments = this.transformToSegments(functions, true);
    segments.splice(0, 0, angular.copy(this.removeSegment));
    return segments;
  });
}
```

alexanderzobnin commented on src/query_builder.js:
```diff
@@ -75,6 +80,15 @@ export default class MQEQuery {
+  addFunctions(functions) {
+    var query = "";
+    if (functions.length) {
+        var result = _.map(functions, function (fn, index) {
```
`_.map()` function usually used for creating new array. If you need to change external variable, like `query` in this case, use `_.forEach()` or native JS `forEach()`:

```
_.forEach(functions, function (fn) {
  query += "|" + fn + " ";
});
return query;
```

alexanderzobnin commented on src/query_ctrl.js:
```diff
@@ -100,6 +103,19 @@ export class MQEQueryCtrl extends QueryCtrl {
   }
 
+  functionSegmentChanged(segment, index) {
+    var _this3 = this;
```
You can also use `let` for variable declaration. `let` prevents hoisting, so variable will block-scoped. See more http://es6-features.org/#BlockScopedVariables

```
let self = this
```

But you don't need to do this with arrow functions.

SeemaKumaran committed:
Review comments incorporated for Functions Implementation

Incorporated review comments for Functions implementation. Also added a
method to refine the functionlist as its not required to show the
operators like *,+,-,/

alexanderzobnin commented on src/query_builder.js:
```diff
@@ -67,6 +67,11 @@ export default class MQEQuery {
 
+      // Render functions if any
+      if(target.functions.length != 0) {
+          query += _this.addFunctions(target.functions);
```
Should be

``` js
query += this.addFunctions(target.functions);
```

alexanderzobnin committed:
Fixed _.this issue and linter errors.

alexanderzobnin committed:
Fixed building query without functions.
